pontoon/base/views.py

```diff
@@ -988,50 +993,75 @@ def transvision(request):
         return HttpResponse("error")
 
 
+def get_download_content(slug, code, relative_path):
+    """
+    Get content for the file to be downloaded.
+
+    :param str slug: Project slug.
+    :param str code: Locale code.
+    :param str relative_path: Resource path.
+    """
+    # Clone or update project repository
+    project = get_object_or_404(Project, slug=slug)
+    project.repositories.first().pull()
+
+    # Get source file path
+    locale = get_object_or_404(Locale, code__iexact=code)
+    vcs_project = VCSProject(project, locales=[locale])
+
+    source_path = relative_path
+    if relative_path.endswith('.po'):
```

Comment:
I wish we had a single common place to put this so that pot files aren't a special case everywhere.